src/Test/Utilities/Portable/Compilation/TestOperationVisitor.cs

```diff
@@ -1120,6 +1120,38 @@ public override void VisitConstantPattern(IConstantPatternOperation operation)
             Assert.Same(operation.Value, operation.Children.Single());
         }
 
+        public override void VisitRelationalPattern(IRelationalPatternOperation operation)
+        {
+            Assert.Equal(OperationKind.RelationalPattern, operation.Kind);
+            VisitPatternCommon(operation);
+            Assert.Same(operation.Value, operation.Children.Single());
+        }
+
+        public override void VisitBinaryPattern(IBinaryPatternOperation operation)
+        {
+            Assert.Equal(OperationKind.BinaryPattern, operation.Kind);
+            VisitPatternCommon(operation);
+            Assert.True(operation.OperatorKind switch { Operations.BinaryOperatorKind.Or => true, Operations.BinaryOperatorKind.And => true, _ => false });
+            var children = operation.Children.ToArray();
+            Assert.Equal(2, children.Length);
+            Assert.Same(operation.LeftPattern, children[0]);
+            Assert.Same(operation.RightPattern, children[1]);
+        }
+
+        public override void VisitNegatedPattern(INegatedPatternOperation operation)
+        {
+            Assert.Equal(OperationKind.NegatedPattern, operation.Kind);
+            VisitPatternCommon(operation);
+            Assert.Same(operation.NegatedPattern, operation.Children.Single());
+        }
+
+        public override void VisitTypePattern(ITypePatternOperation operation)
```

Comment:
>ITypePatternOperation [](start = 46, length = 21)

Consider asserting ```MatchedType``` is not null.

Reply:
Will address in [https://github.com/dotnet/roslyn/pull/42548](https://github.com/dotnet/roslyn/pull/42548)

---
In reply to: [394690849](https://github.com/dotnet/roslyn/pull/42474#discussion_r394690849) [](ancestors = 394690849)